### ISSUE TITLE
labextension install fails with Python 2.7: "'module' object has no attribute 'DEVNULL'"

### ISSUE THREAD

**Issue by NathanDunfield** [jupyterlab/jupyterlab#5129]
Setup:  JupyterLab v0.33.11, Ubuntu 16.04, Python 2.7.15 (built from source), node 8.11.3.  

Problem: 

```
% jupyter labextension install --debug @jupyter-widgets/jupyterlab-manager
Searching [u'/home/sage', '/home/sage/.sage//jupyter-4.1', '/sage/local/etc/jupyter', '/usr/local/etc/jupyter', '/etc/jupyter'] for config files
Looking for jupyter_config in /etc/jupyter
Looking for jupyter_config in /usr/local/etc/jupyter
Looking for jupyter_config in /sage/local/etc/jupyter
Looking for jupyter_config in /home/sage/.sage//jupyter-4.1
Looking for jupyter_config in /home/sage
Traceback (most recent call last):

  File "/sage/local/lib/python2.7/site-packages/jupyterlab/labextensions.py", line 79, in start
    ans = self.run_task()

  File "/sage/local/lib/python2.7/site-packages/jupyterlab/labextensions.py", line 108, in run_task
    for arg in self.extra_args

  File "/sage/local/lib/python2.7/site-packages/jupyterlab/commands.py", line 227, in install_extension
    _node_check()

  File "/sage/local/lib/python2.7/site-packages/jupyterlab/commands.py", line 1466, in _node_check
    proc = Process(['node', 'node-version-check.js'], cwd=HERE, quiet=True)

  File "/sage/local/lib/python2.7/site-packages/jupyterlab_launcher/process.py", line 109, in __init__
    kwargs['stdout'] = subprocess.DEVNULL

AttributeError: 'module' object has no attribute 'DEVNULL'
```

Likely cause: In Python 2.7, the subprocess module does not have a DEVNULL attribute, this was added in Python 3.3.

**Comment by tfriedel** [comment]
this can be solved by installing / upgrading the backport of python 3 subprocess:
pip install --upgrade subprocess32

Since jupyterlab depends on this the requirments should be updated.

**Comment by blink1073** [comment]
We had previously `subprocess32` in `setup.py` as a [requirement](https://github.com/jupyterlab/jupyterlab/blob/0ce5798b5f0993db7e6aeca8bb94f2c6907c37fd/setup.py#L137) for Python 2.  In general, supporting Python 2 has become too much of a maintenance burden, and we are no longer supporting it.  cf #2096.

**Comment by blink1073** [comment]
We'd accept a PR against 0.33.x that addresses the above.

**Comment by NathanDunfield** [comment]
@blink1073 So `subprocess32` is still a requirement for Python 2, but now you need version 3.5.x of that module; I only had the previous version (3.2.7) which is why I got the error.  I created a pull request.

**Comment by blink1073** [comment]
Fixed by #5140 and released in `0.33.12`.  Thanks!
